### ISSUE TITLE
Bootstrap sass imports result in error

### ISSUE THREAD

**Issue by Julioevm** [parcel-bundler/parcel#10010]
# 🐛 bug report

scss import statements from bootstrap result in a:

`  Error: The "path" argument must be of type string or an instance of URL. Received null`

## 🎛 Configuration (.babelrc, package.json, cli command)

```js
{
  "name": "parcel-bootstrap-issue",
  "version": "1.0.0",
  "description": "reproduction for an issue with parcel 2.13.0",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "build": "parcel build 'src/*.html'"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "bootstrap": "^5.3.3",
    "parcel": "2.13.0",
    "@parcel/transformer-sass": "2.13.0"
  }
}
```

## 🤔 Expected Behavior

Necessary imports are bundled correctly.

## 😯 Current Behavior

bootstrap scss import statements result in a:

`  Error: The "path" argument must be of type string or an instance of URL. Received null`

```⠇ Building style.scss...

**omiting some deprecation warnings**

🚨 Build failed.

@parcel/transformer-sass: The "path" argument must be of type string or an instance of URL. Received null
  ╷
1 │ @import "mixins/banner";
  │         ^^^^^^^^^^^^^^^
  ╵
  node_modules/bootstrap/scss/bootstrap-reboot.scss 1:9  @use
  src/style.scss 1:1                                     root stylesheet

  Error: The "path" argument must be of type string or an instance of URL. Received null
    ╷
  1 │ @import "mixins/banner";
    │         ^^^^^^^^^^^^^^^
    ╵
    node_modules/bootstrap/scss/bootstrap-reboot.scss 1:9  @use
    src/style.scss 1:1                                     root stylesheet
      at Object.wrapException (/Users/julio.valls/repos/parcel-bootstrap-issue/node_modules/sass/sass.dart.js:2305:43)
      at Object.throwWithTrace0 (/Users/julio.valls/repos/parcel-bootstrap-issue/node_modules/sass/sass.dart.js:32212:15)
      at /Users/julio.valls/repos/parcel-bootstrap-issue/node_modules/sass/sass.dart.js:87502:19
      at _wrapJsFunctionForAsync_closure.$protected (/Users/julio.valls/repos/parcel-bootstrap-issue/node_modules/sass/sass.dart.js:5020:15)
      at _wrapJsFunctionForAsync_closure.call$2 (/Users/julio.valls/repos/parcel-bootstrap-issue/node_modules/sass/sass.dart.js:37950:12)
      at _awaitOnObject_closure0.call$2 (/Users/julio.valls/repos/parcel-bootstrap-issue/node_modules/sass/sass.dart.js:37944:25)
      at Object._rootRunBinary (/Users/julio.valls/repos/parcel-bootstrap-issue/node_modules/sass/sass.dart.js:5437:18)
      at StaticClosure.<anonymous> (/Users/julio.valls/repos/parcel-bootstrap-issue/node_modules/sass/sass.dart.js:123976:16)
      at _CustomZone.runBinary$3$3 (/Users/julio.valls/repos/parcel-bootstrap-issue/node_modules/sass/sass.dart.js:39404:39)
      at _FutureListener.handleError$1 (/Users/julio.valls/repos/parcel-bootstrap-issue/node_modules/sass/sass.dart.js:38167:21)

```

## 🔦 Context

We use some bootstrap sass stuff in our application, this was working fine in parcel 2.12.0 but trying to update results in the described issue.

## 💻 Code Sample

Simplified reproduction of the issue:

https://github.com/Julioevm/parcel-bootstrap-issue

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.13.0
| Node             | 21.4.0
| npm/Yarn         | 10.2.4
| Operating System | Mac OS Sonoma (silicon) / Reproducible on CI linux container

**Comment by sumo** [comment]
Can confirm it fails with bulma and a custom stylesheet

`@parcel/transformer-sass: The "path" argument must be of type string or an instance of URL. Received null
  ╷
4 │ @use "sass";
  │ ^^^^^^^^^^^
  ╵
  ../node_modules/bulma/bulma.scss 4:1  @import
  src/styles.scss 1:9                   root stylesheet

  Error: The "path" argument must be of type string or an instance of URL. Received null
    ╷
  4 │ @use "sass";
    │ ^^^^^^^^^^^
    ╵
    ../node_modules/bulma/bulma.scss 4:1  @import
    src/styles.scss 1:9                   root stylesheet
      at Object.wrapException (/home/sumit/dev/matrixid-web/node_modules/sass/sass.dart.js:2305:43)
      at Object.throwWithTrace0 (/home/sumit/dev/matrixid-web/node_modules/sass/sass.dart.js:32212:15)
      at /home/sumit/dev/matrixid-web/node_modules/sass/sass.dart.js:87502:19
      at _wrapJsFunctionForAsync_closure.$protected (/home/sumit/dev/matrixid-web/node_modules/sass/sass.dart.js:5020:15)
      at _wrapJsFunctionForAsync_closure.call$2 (/home/sumit/dev/matrixid-web/node_modules/sass/sass.dart.js:37950:12)
      at _awaitOnObject_closure0.call$2 (/home/sumit/dev/matrixid-web/node_modules/sass/sass.dart.js:37944:25)
      at Object._rootRunBinary (/home/sumit/dev/matrixid-web/node_modules/sass/sass.dart.js:5437:18)
      at StaticClosure.<anonymous> (/home/sumit/dev/matrixid-web/node_modules/sass/sass.dart.js:123976:16)
      at _CustomZone.runBinary$3$3 (/home/sumit/dev/matrixid-web/node_modules/sass/sass.dart.js:39404:39)
      at _FutureListener.handleError$1 (/home/sumit/dev/matrixid-web/node_modules/sass/sass.dart.js:38167:21)`

**Comment by theweinzierl** [comment]
I have the same issue when upgrading to parcel 2.13.0, using a similar configuration like @Julioevm and @sumo.

**Comment by bacchilu** [comment]
I confirm the same error described above with _Parcel 2.13.0_.

**Comment by mariots** [comment]
I am also facing the same issue when upgrading to parcel 2.13.0

**Comment by shanekunz** [comment]
Just confirming same issue with bootstrap. using ~2.12 instead of ^2 for now.

**Comment by Vresod** [comment]
Confirming same issue on parcel 2.13.0 and bootstrap 5.3.3.

**Comment by jord4nrowe** [comment]
Hello having the same issue with parcel 2.13.0 and zurb foundation 6.9.0.

Also on Apple Silicon

**Comment by devongovett** [comment]
What version of sass do you have installed?

**Comment by jtojnar** [comment]
I can also reproduce this with the [latest `npm update`](https://github.com/fossar/selfoss/pull/1506), which bumped `sass` to `1.81.0`.

Downgrading the `sass` version to the previous one with `npm install --prefix client 'sass@1.77.6' --save-dev` did get rid of the `@import` deprecation warning:

```
DEPRECATION WARNING: Sass @import rules are deprecated and will be removed in Dart Sass 3.0.0.

More info and automated migrator: https://sass-lang.com/d/import

  ╷
4 │ @import 'npm:reset-css/sass/reset';
  │         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  ╵
    styles/main.scss 4:9  root stylesheet
```

But the error from the OP remained:

```
@parcel/transformer-sass: The "path" argument must be of type string or an instance of URL. Received null
  ╷
4 │ @import 'npm:reset-css/sass/reset';
  │         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  ╵
```

**Comment by devongovett** [comment]
Should be fixed by #10018. Note that you can suppress the SASS deprecation warnings rather than downgrading it. Add this to your `.sassrc`:

```json
{
  "silenceDeprecations": ["import"]
}
```

**Comment by jtojnar** [comment]
Thanks, this helped but now I see a different error: https://github.com/parcel-bundler/parcel/issues/10019
